ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/UserServiceContext.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.server;
+
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.ksql.services.DefaultServiceContext;
+import io.confluent.ksql.services.KafkaTopicClient;
+import io.confluent.ksql.services.ServiceContext;
+import io.confluent.ksql.util.KsqlConfig;
+import java.util.function.Supplier;
+import javax.ws.rs.core.SecurityContext;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.streams.KafkaClientSupplier;
+
+
+public final class UserServiceContext implements ServiceContext {
+  public static UserServiceContext create(
+      final SecurityContext securityContext,
+      final KsqlConfig ksqlConfig
+  ) {
+    if (securityContext.getUserPrincipal() == null) {
+      return new UserServiceContext(DefaultServiceContext.create(ksqlConfig));
+    }
+
+    // TODO: Create a UserServiceContext using user credentials
```

Comment:
Since this class isn't used in this PR let's introduce it in a future one.

Reply:
Oh, mmm, I left this code from another patch I have. I was splitting the other patch with only the refactor, and I forgot to remove it. Thanks for catching this.